SecurityService.OpenIdConnect.IntegrationTests/Common/DockerHelper.cs

```diff
@@ -1,137 +1,327 @@
+    using Shared.Logger;
 
-    public class DockerHelper
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <seealso cref="Shared.IntegrationTesting.DockerHelper" />
+    public class DockerHelper : Shared.IntegrationTesting.DockerHelper
     {
-        public INetworkService TestNetwork;
+        #region Fields
+
+        /// <summary>
+        /// The security service client
+        /// </summary>
+        public ISecurityServiceClient SecurityServiceClient;
 
-        public IContainerService SecurityServiceTestUIContainer;
+        /// <summary>
+        /// The security service container name
+        /// </summary>
+        public String SecurityServiceContainerName;
 
+        /// <summary>
+        /// The security service test UI container name
+        /// </summary>
         public String SecurityServiceTestUIContainerName;
 
-        public ISecurityServiceClient SecurityServiceClient;
-        
+        /// <summary>
+        /// The security service test UI port
+        /// </summary>
+        public Int32 SecurityServiceTestUIPort;
+
+        /// <summary>
+        /// The test identifier
+        /// </summary>
         public Guid TestId;
-        private void SetupTestNetwork()
+
+        /// <summary>
+        /// The containers
+        /// </summary>
+        protected List<IContainerService> Containers;
+
+        /// <summary>
+        /// The security service port
+        /// </summary>
+        protected Int32 SecurityServicePort;
+
+        /// <summary>
+        /// The test networks
+        /// </summary>
+        protected List<INetworkService> TestNetworks;
+
+        /// <summary>
+        /// The logger
+        /// </summary>
+        private readonly NlogLogger Logger;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DockerHelper"/> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        public DockerHelper(NlogLogger logger)
         {
-            // Build a network
-            this.TestNetwork = new Ductus.FluentDocker.Builders.Builder().UseNetwork($"testnetwork{this.TestId}").Build();
+            this.Logger = logger;
+            this.Containers = new List<IContainerService>();
+            this.TestNetworks = new List<INetworkService>();
         }
 
-        public async Task StartContainersForScenarioRun(String scenarioName)
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Starts the containers for scenario run.
+        /// </summary>
+        /// <param name="scenarioName">Name of the scenario.</param>
+        public override async Task StartContainersForScenarioRun(String scenarioName)
         {
-            String traceFolder = $"/home/txnproc/trace/{scenarioName}/";
+            String traceFolder = FdOs.IsWindows() ? $"C:\\home\\txnproc\\trace\\{scenarioName}" : $"/home/txnproc/trace/{scenarioName}";
```

Comment:
Codacy found an issue: [Remove the unused local variable 'traceFolder'.](https://app.codacy.com/manual/stuart_ferguson1/SecurityService/pullRequest?prid=6144581)